integration/spark-common/src/main/scala/org/apache/spark/sql/catalyst/CarbonDDLSqlParser.scala

```diff
@@ -787,6 +787,19 @@ abstract class CarbonDDLSqlParser extends AbstractCarbonSparkSQLParser {
       }
     }
 
+    // range_column should be there in create table cols
+    if (tableProperties.get(CarbonCommonConstants.RANGE_COLUMN).isDefined) {
+      val rangeColumn = tableProperties.get(CarbonCommonConstants.RANGE_COLUMN).get.trim
+      val rangeField = fields.find(_.column.equalsIgnoreCase(rangeColumn))
+      if (rangeField.isEmpty) {
+        val errorMsg = "range_column: " + rangeColumn +
+                       " does not exist in table. Please check the create table statement."
+        throw new MalformedCarbonCommandException(errorMsg)
+      } else {
+        tableProperties.put(CarbonCommonConstants.RANGE_COLUMN, rangeField.get.column)
```

Comment:
So 
1. range_column can be all kinds of datatype?
2. range_column can be non-sort_column?

Reply:
both  yes